lib/internal/util.js

```diff
@@ -18,7 +18,11 @@ exports.printDeprecationMessage = function(msg, warned) {
 };
 
 exports.error = function(msg) {
-  console.error(`${prefix}${msg}`);
+  if (arguments.length > 1) {
+    arguments[0] = `${prefix}${msg}`;
```

Comment:
I thought so too, but according to @petkaantonov's ["Optimization killers" page](https://github.com/petkaantonov/bluebird/wiki/Optimization-killers#3-managing-arguments) it doesn't apply if the script/function is in strict mode?